From 9d2e3d3287fcdaed56b9084f9c332ddb92962f0d Mon Sep 17 00:00:00 2001
From: jschyra <jschyra@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 7 Aug 2012 11:14:24 +0000
Subject: [PATCH 0307/1076]  - Der LAN Interface Name kann jetzt auch von der
 INI Datei gesetzt werden.  - Konfigurationswerte in der INI Datei
 Dokumentiert.

git-svn-id: svn://localhost/SelfServiceCommon/trunk@373 90b65887-3827-0410-9a23-83215b262276
---
 .../Build/setup/SetKioskData/SetKioskData.ini | 20 +++++++++++++++++++
 .../Build/setup/SetKioskData/SetKioskData.iss | 11 +++++++---
 2 files changed, 28 insertions(+), 3 deletions(-)

diff --git a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.ini b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.ini
index 028f89cc..ef46db2c 100644
--- a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.ini
+++ b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.ini
@@ -1,13 +1,33 @@
+; Kiosk data configuration section.
 [KIOSKDATA]
+; Kiosk environment KIOSK_NAME Default. 
+; The KIOSK_NAME will also be used for the new windows hostname. 
 Kiosk Name:=CUSS-TEMP
+; Kiosk environment AIRPORT Default
 Airport: (3L Code)=DTM
+; Kiosk environment AREA Default
 Area: (Area in Terminal)=CheckIn
+; Kiosk environment TERMINAL Default
 Terminal:=A
+; The application to configure. NOTE: more then one applications are not supportet yet.
 Applications=MAT_CKI
+
+; Host data configuration section.
 [HOSTDATA]
+; Image host- (and kiosk-)name.
+; The setup need this to change the host and kiosk name to the new name. 
+; You can also set a part of the host name, if the setup should only cange this. 
 ImageHostName=KioskTemplate
+; The name of the LAN interface to edit. Default: Local Area Connection 
+IPv4Interface=LAN-Verbindung
+; Default IPv4 address for the GUI
 IPv4Address=139.2.249.21
+; Default IPv4 subnet for the GUI
 IPv4Subnet=255.255.240.0
+; Default IPv4 Gateway for the GUI
 IPv4Gateway=139.2.240.1
+
+; Configuration section for the Kioskapplication MAT_CKI
 [APPDATA_MAT_CKI]
+; The application command line string.
 CMD="C:\Programme\Internet Explorer\iexplore.exe" -K http://cuss-mat-01/Applications/MAT/MAT_App.html$ wnd=MAT_App
diff --git a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
index f6b0728a..e4af2b4d 100644
--- a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
+++ b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
@@ -82,9 +82,9 @@ Name: {commonstartup}\SetKioskData.lnk; Type: files
 Name: {userstartup}\SetKioskData.lnk; Type: files
 
 [Run]
-Filename: {sys}\netsh; Parameters: "interface ip set address name=""LAN-Verbindung"" source=static addr={code:GetIPv4Address} mask={code:GetIPv4Subnet}"; WorkingDir: {sys}; Flags: runhidden
-Filename: {sys}\netsh; Parameters: "interface ip set address name=""LAN-Verbindung"" gateway={code:GetIPv4Gateway} gwmetric=0"; WorkingDir: {sys}; Flags: runhidden
-Filename: {sys}\netsh; Parameters: "interface ip set dns name=""LAN-Verbindung"" source=static addr=none"; WorkingDir: {sys}; Flags: runhidden
+Filename: {sys}\netsh; Parameters: "interface ip set address name=""{code:GetLANInterfaceName}"" source=static addr={code:GetIPv4Address} mask={code:GetIPv4Subnet}"; WorkingDir: {sys}; Flags: runhidden
+Filename: {sys}\netsh; Parameters: "interface ip set address name=""{code:GetLANInterfaceName}"" gateway={code:GetIPv4Gateway} gwmetric=0"; WorkingDir: {sys}; Flags: runhidden
+Filename: {sys}\netsh; Parameters: "interface ip set dns name=""{code:GetLANInterfaceName}"" source=static addr=none"; WorkingDir: {sys}; Flags: runhidden
 [Code]
 var
 
@@ -122,6 +122,11 @@ begin
 	Result := IPv4LevelData[2];
 end;
 
+function GetLANInterfaceName(Param:String) : String
+begin
+    Result := ProcessIniString('HOSTDATA', 'IPv4Interface', 'Local Area Connection');
+end;
+
 function UpdateReadyMemo(Space, NewLine, MemoUserInfoInfo, MemoDirInfo, MemoTypeInfo, MemoComponentsInfo, MemoGroupInfo, MemoTasksInfo: String): String;
 var
   S: String;
-- 
2.41.0.windows.1

